src/js/libcs/Accessors.js

```diff
@@ -168,6 +168,22 @@ Accessor.setField = function(geo, field, value) {
         movepointscr(geo, List.turnIntoCSList([value.value[0], value.value[1], CSNumber.real(1)]), "homog");
     }
 
+    if (field === "x" && geo.kind === "P" && geo.movable && value.ctype === "number") {
+        if (CSNumber._helper.isAlmostZero(geo.homog.value[2])) {
+            console.log("cannot set x component without y component of far points");
+        } else {
+            movepointscr(geo, List.turnIntoCSList([value, CSNumber.div(geo.homog.value[1], geo.homog.value[2]), CSNumber.real(1)]), "homog");
+        }
+    }
```

Comment:
Beware that if `y` and `z` are zero, we assign `[0, 0, 0]` as homogeneous coordinates. Is this OK?